dev/TabView/TabView.cpp

```diff
@@ -233,14 +233,42 @@ void TabView::OnListViewLoaded(const winrt::IInspectable&, const winrt::RoutedEv
         // Now that ListView exists, we can start using its Items collection.
         auto items = TabItems();
         auto numItemsToCopy = static_cast<int>(items.Size());
-        if (auto lvItems = listView.Items())
+        bool itemsSourceDoesNotExists = listView.ItemsSource() == nullptr;
+
+        if (itemsSourceDoesNotExists)
+        {
+            if (auto lvItems = listView.Items())
+            {
+                // copy the list, because clearing lvItems may also clear items
+                winrt::IVector<winrt::IInspectable> item_list{ winrt::single_threaded_vector<winrt::IInspectable>() };
+
+                for (auto itm : items)
+                {
+                    item_list.Append(itm);
+                }
+
+                lvItems.Clear();
+
+                for (int i = 0; i < numItemsToCopy; i++)
+                {
+                    // App put items in our Items collection; copy them over to ListView.Items
+                    auto itm = item_list.GetAt(i);
+                    if (itm != nullptr)
+                        lvItems.Append(itm);
+                }
+                TabItems(lvItems);
+            }
+        }
+
+        else if (auto itemsSource = listView.ItemsSource().try_as<winrt::IIterable<winrt::IInspectable>>())
         {
-            for (int i = 0; i < numItemsToCopy; i++)
+            winrt::IVector<winrt::IInspectable> item_list{ winrt::single_threaded_vector<winrt::IInspectable>() };
+
+            for (auto itm : itemsSource)
             {
-                // App put items in our Items collection; copy them over to ListView.Items
-                lvItems.Append(items.GetAt(i));
+                item_list.Append(itm);
             }
-            TabItems(lvItems);
+            TabItems(item_list);
         }
```

Comment:
Why is this needed? can't we just do `TabItems(listView.ItemsSource())` since at this point we are sure that the items source exists

Reply:
I wish I could do that, but apparently it creates an error since the `TabItems` property only accepts `IVector<>` and Idon't know what type of object is that `ItemsSource` . So I had to make sure it's iterable , then make an `IVector<>` from it and pass it as the parameter.

Reply:
I have deleted the else block, since it's unnecessary.